### PR TITLE
Document Generator fix to correct abridged Json structure for ixbrl

### PR DESCRIPTION
**Pull request by Mathsh**
Added new AbridgedAccountsApiData model
Updated AbridgedAccountsDataHanlder to use objectMapper to convert from api data to json

### CODE REVIEW

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java [hunk 1]
```diff
@@ -142,21 +147,21 @@ private String createPathString(AccountType accountType) {
...
+        abridgedAccountsApiData.setAbridgedAccountsApi(accountData);
 
         CompanyProfileApi companyProfile = companyService.getCompanyProfile(transaction.getCompanyNumber());
-        account.put("company_name", companyProfile.getCompanyName());
+        abridgedAccountsApiData.setCompanyName(companyProfile.getCompanyName());
+        abridgedAccountsApiData.setCompanyNumber(companyProfile.getCompanyNumber());
+
+        ObjectMapper mapper = new ObjectMapper();
+        String accountsJSON = mapper.writeValueAsString(abridgedAccountsApiData);
+        JsonNode accounts = mapper.readTree(accountsJSON);
```
Comment (ch-mcabrera) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java]: We could move this block  to different method following the Single Responsibility Principle.  
```
ObjectMapper mapper = new ObjectMapper();
String accountsJSON = mapper.writeValueAsString(abridgedAccountsApiData);
JsonNode accounts = mapper.readTree(accountsJSON);
```

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/abridged/model/AbridgedAccountsApiData.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if (!(o instanceof AbridgedAccountsApiData)) return false;
+        AbridgedAccountsApiData that = (AbridgedAccountsApiData) o;
+        return Objects.equals(getAbridgedAccountsApi(), that.getAbridgedAccountsApi()) &&
+                Objects.equals(getCompanyNumber(), that.getCompanyNumber()) &&
+                Objects.equals(getCompanyName(), that.getCompanyName());
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (ch-mcabrera) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/abridged/model/AbridgedAccountsApiData.java]: empty line ;)

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java (hunk 1 above)
Comment (Mathsh) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java]: done

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/abridged/model/AbridgedAccountsApiData.java (hunk 2 above)
Comment (Mathsh) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/abridged/model/AbridgedAccountsApiData.java]: done

**Pull request merged**